front/src/modules/simulationResult/components/ManchetteWithSpaceTimeChart/ManchetteWithSpaceTimeChart.tsx

```diff
@@ -100,16 +118,41 @@ const ManchetteWithSpaceTimeChartWrapper = ({
 
   const [waypointsPanelIsOpen, setWaypointsPanelIsOpen] = useState(false);
 
-  const [tmpSelectedTrain, setTmpSelectedTrain] = useState(selectedTrainScheduleId);
-
-  useEffect(() => {
-    setTmpSelectedTrain(selectedTrainScheduleId);
-  }, [selectedTrainScheduleId]);
+  const projectedTrainsAndOccurrences = useMemo(
+    () =>
+      projectPathTrainResult.flatMap((train) => {
+        if (isOccurrence(train.id) && train.paced) {
+          const pacedTrainId = getPacedTrainIdFromOccurrenceId(train.id);
+          const occurrencesCount = Math.ceil(
+            Duration.parse(train.paced.duration).ms / Duration.parse(train.paced.step).ms
+          );
+          const occurrences = [];
+          for (let i = 0; i < occurrencesCount; i += 1) {
+            const occurrenceStartTime = dayjs(train.departureTime)
+              .add(i * Duration.parse(train.paced.step).ms, 'ms')
```

Comment:
nit: you can create a const stepInMs outside of the for loop and reuse it here
```suggestion
              .add(i * stepInMs, 'ms')
```

Reply:
done